### PR TITLE
Update some sub-projects to 3.3.0

### PR DESCRIPTION
**Pull request by s-bernard**
Simple update of rocrand, rocprim, hipsparse, rocm-clang-ocl (I included also clang-ocl) to 3.3.0. I needed them to compile PyTorch.

No need for a patch for hipsparse anymore.

### CODE REVIEW

File: clang-ocl/PKGBUILD [hunk 1]
```diff
@@ -1,14 +1,14 @@
 # Maintainer: Jakub Okoński <jakub@okonski.org>
 pkgname=clang-ocl
-pkgver=3.1.0
+pkgver=3.3.0
```
Comment (acxz) [inline clang-ocl/PKGBUILD:3]: Do you mind if do not upgrade clang-ocl? This package is slated to be removed, no need to continue development on it.

File: hipsparse/PKGBUILD [hunk 2]
```diff
@@ -28,21 +20,18 @@ build() {
...
-        -Drocsparse_DIR=/opt/rocm/rocsparse/lib/cmake/rocsparse \
+        -DCMAKE_INSTALL_PREFIX=/opt/rocm/hipsparse \
         -DBUILD_CLIENTS_SAMPLES=OFF \
         -DBUILD_CLIENTS_TESTS=OFF \
-        -DHSA_HEADER=/opt/rocm/hsa/include \
+        -Drocsparse_DIR=/opt/rocm/rocsparse/lib/cmake/rocsparse \
         "$srcdir/hipSPARSE-rocm-$_pkgver"
-  make
+
+  make -j $(nproc)
```
Comment (acxz) [inline hipsparse/PKGBUILD:34]: same issue as before

File: rocm-clang-ocl/PKGBUILD [hunk 3]
```diff
@@ -19,8 +19,9 @@ build() {
   cmake -DCMAKE_BUILD_TYPE=Release \
         -DOPENCL_ROOT=/opt/rocm/hcc \
         -DCMAKE_INSTALL_PREFIX=/opt/rocm \
-        "$srcdir/$_pkgname-roc-$pkgver"
-  make
+        "$srcdir/$_pkgname-rocm-$pkgver"
+
+  make -j $(nproc)
```
Comment (acxz) [inline rocm-clang-ocl/PKGBUILD:24]: same issue as before

File: rocprim/PKGBUILD [hunk 4]
```diff
@@ -29,7 +29,8 @@ build() {
         -DBUILD_TEST=OFF \
         -Damd_comgr_DIR=/opt/rocm/lib/cmake/amd_comgr \
         "$srcdir/rocPRIM-rocm-$pkgver"
-  make
+
+  make -j $(nproc)
```
Comment (acxz) [inline rocprim/PKGBUILD:33]: same issue as before

File: rocrand/PKGBUILD [hunk 5]
```diff
@@ -22,8 +22,9 @@ build() {
   cmake -DCMAKE_BUILD_TYPE=Release \
         -DBUILD_TEST=OFF \
         -DCMAKE_CXX_COMPILER=/opt/rocm/hcc/bin/hcc \
-        "$srcdir/rocRAND-$pkgver"
-  make
+        "$srcdir/rocRAND-rocm-$pkgver"
+
+  make -j $(nproc)
```
Comment (acxz) [inline rocrand/PKGBUILD:27]: same issue as before

File: clang-ocl/PKGBUILD (hunk 1 above)
Comment (s-bernard) [reply clang-ocl/PKGBUILD:3]: Not at all, I did it first, that's why I included it here. I'm removing it.

File: rocm-clang-ocl/PKGBUILD [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
-pkgver=3.1.0
-pkgrel=2
+pkgver=3.3.0
+pkgrel=1
 pkgdesc="OpenCL compilation with clang compiler."
 arch=('x86_64')
 url="https://github.com/RadeonOpenCompute/clang-ocl"
 license=('unknown')
 makedepends=('cmake' 'hcc' 'rocm-cmake')
-source=("${pkgname}-${pkgver}::http://github.com/RadeonOpenCompute/clang-ocl/archive/roc-${pkgver}.tar.gz")
```
Comment (acxz) [inline rocm-clang-ocl/PKGBUILD:11]: Why was `${pkgname}-${pkgver} removed? It is good practice to uniquely name the downloaded tar.gz

Comment (s-bernard) [reply rocm-clang-ocl/PKGBUILD:11]: No reason, I did know this convention and did not find it in every package. I can put it back.

**Pull request merged**